portfolio/src/main/webapp/script.js

```diff
@@ -83,9 +83,98 @@ for (i = 0; i < coll.length; i++) {
     });
 }
 
+var markers = [];
+
+var mich_marker;
```

Comment:
I think this can be a local variable that you pass into relevant methods, rather than having a global variable.